pmd-core/src/main/java/net/sourceforge/pmd/autofixes/operations/PlaceholderTextOperation.java

```diff
@@ -0,0 +1,29 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.autofixes.operations;
+
+import net.sourceforge.pmd.autofixes.Document;
+
+/**
+ * This text operation type is intended to mark the total region of text operations and will apply
+ * all the text operations of its children over the document, but this instance will not apply any
+ * operation over the document.
+ */
+public class PlaceholderTextOperation extends TextOperation {
```

Comment:
The placeholderOperation defines a region from which all its children operations will be done. As the operations are modeled in a tree manner, if i have a list of children who represent a set of changes, I would not always want to apply a change in the root node which covers all the region after it is applied to my children.

Reply:
So... this will **always** be the root text operation? If so...
1. the name should reflect this better
2. you shouldn't be able to freely create new instances... maybe have the document create and hold one internally?
3. instead of freely adding a text operation to another, you could just add operations to the document, meaning you don't need to keep the root of the tree yourself...

Just thinking out loud here

Reply:
That's right, this will always be the root text operation and we shouldn't be able to freely create new instances, however we currently don't have any class in which to delegate the addition of edits to the tree. I put some thought on adding the operations to the document but that would increase coupling between the Document implementation and the way the text operations are applied, having two reasons to exist, the representation of a source file and the changes it holds.

Reply:
True, yet there is an inherent coupling between them both, as the root defines a range equals to the document length. Tat also means, the same tree CAN'T be simply applied to a different document, but is tied to a single one. This **can** be on the side of the operation, but still, the one-on-one relationship and the binding between doc and operations must be enforced.

Also, the operations is setup as a tree, yet I see no logic to build this tree (just adding nodes to a parent, but not a logic on what nodes should be children of which), so I'm not sure how this would work out, or if a tree structure is even needed.